pkg/operatorstatus/operator_status.go

```diff
@@ -243,24 +262,49 @@ func printOperandVersions(versions []configv1.OperandVersion) string {
 	return strings.Join(versionsOutput, ", ")
 }
 
-func clusterObjectNeedsUpdating(co *configv1.ClusterOperator, conds []configv1.ClusterOperatorStatusCondition, desiredVersions []configv1.OperandVersion, desiredRelatedObjects []configv1.ObjectReference) (*configv1.ClusterOperator, bool) {
-	shouldUpdate := false
+// isRelatedObjectsDeepEqual compares two slices of ObjectReference and returns true if they are equal.
+// Slices that have the same elements but different ordering are still considered equal.
+func isRelatedObjectsDeepEqual(currentRelatedObjects, desiredRelatedObjects []configv1.ObjectReference) bool {
+	// Deep copy current related objects to avoid modifying the original slice.
+	currentRelatedObjectsCopy := make([]configv1.ObjectReference, len(currentRelatedObjects))
+	copy(currentRelatedObjectsCopy, currentRelatedObjects)
 
-	for _, cond := range conds {
-		if !v1helpers.IsStatusConditionPresentAndEqual(co.Status.Conditions, cond.Type, cond.Status) {
-			shouldUpdate = true
-		}
-	}
+	// Deep copy desired related objects to avoid modifying the original slice.
```

Comment:
Is ordering an undesired mutation? I wouldn't think it would matter that much, since the additional memory usage is negligible.